libmcount/mcount.c

```diff
@@ -543,6 +544,9 @@ int mcount_entry(unsigned long *parent_loc, unsigned long child,
 	/* fixup the parent_loc in an arch-dependant way (if needed) */
 	parent_loc = mcount_arch_parent_location(&symtabs, parent_loc, child);
 
+	/* python hooking for function entry */
+	python_mcount_entry(child, *parent_loc);
```

Comment:
This also needs to be generic - some like `mcount_script_entry()`.  And I think it'd be better passing more info like tid, time, symbol name, depth and so on.